core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/CountDistinct.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.batch.aggregator.function;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Count distinct values of a specific column
+ */
+public class CountDistinct implements AggregateFunction<Integer> {
+  private final String fieldName;
+  private Set<Object> collectSet;
```

Comment:
this will only work for some field types. Those types should be validated. More specifically, this will not "work" except on ints, longs, strings, floats, and doubles. And for floats and doubles, it doesn't really make much sense.

If it needs to work for other types, it is quite a bit more complex.

Reply:
@albertshau can you explain why we'd need to validate specific types? most of the use cases I can think of would be satisfied with strings, ints, longs, floats and doubles. Perhaps booleans in addition. But I'm unable to follow why it wouldn't work for others, and by others you mean the other types in the Schema object, right?

Reply:
fixed